src/main/java/duke/task/DeadlineTask.java

```diff
@@ -0,0 +1,33 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A task that should be completed before the specified date.
+ */
+public class DeadlineTask extends Task {
+    private LocalDate d;
```

Comment:
Consider using a more descriptive name for this attribute.
```suggestion
    private LocalDate date;
```